assets/pages/oneoff-contributions/helpers/ajax.js

```diff
@@ -92,18 +92,18 @@ function postToEndpoint(request: Object, dispatch: Function, abParticipations: P
       dispatch(checkoutSuccess());
     }
     return response.json();
-  }).then((responseJson) => {
-    if (responseJson.error.exceptionType === 'CardException') {
-      dispatch(checkoutError('Your card has been declined.'));
-    } else {
-      const errorHttpCode = responseJson.error.errorCode || 'unknown';
-      const exceptionType = responseJson.error.exceptionType || 'unknown';
-      const errorName = responseJson.error.errorName || 'unknown';
-      logException(`Stripe payment attempt failed with following error: code: ${errorHttpCode} type: ${exceptionType} error-name: ${errorName}.`);
-      dispatch(checkoutError('There was an error processing your payment. Please\u00a0try\u00a0again\u00a0later.'));
+  }).then((data) => {
+    if (data.type === 'error') {
+      const { error: { exceptionType, responseCode, errorName = 'Unknown' } } = data;
```

Comment:
If `error` is `undefined` we'll get an exception, something like:
```
Uncaught TypeError: Cannot destructure property `exceptionType` of 'undefined' or 'null'.
```
This will handled by the final catch. Checking that's ok / intended?

Reply:
Hi Guy, 
Thats a good point, the catch will only show a error for the failure of the function itself, rather than the unexpected response that caused the failure. Probably better to handle the absence of expected fields in the de-structure step. Maybe adding default values?